common/persistence/nosql/nosql_task_store_test.go

```diff
@@ -104,3 +109,175 @@ func TestGetTaskListSize(t *testing.T) {
+	assert.Equal(t, TestDomainID, taskListInfo.DomainID)
+	assert.Equal(t, TestTaskListName, taskListInfo.Name)
+	assert.Equal(t, int(types.TaskListTypeDecision), taskListInfo.TaskType)
+	assert.Equal(t, initialRangeID, taskListInfo.RangeID)
+	assert.Equal(t, initialAckLevel, taskListInfo.AckLevel)
+	assert.Equal(t, int(types.TaskListKindNormal), taskListInfo.Kind)
+	assert.WithinDuration(t, time.Now(), taskListInfo.LastUpdated, time.Second)
+}
+
+func TestLeaseTaskList_emptyTaskList(t *testing.T) {
+	store, _ := setupNoSQLStoreMocks(t)
+
+	req := getValidLeaseTaskListRequest()
+	req.TaskList = ""
+	_, err := store.LeaseTaskList(ctx.Background(), req)
+
+	var expectedErr *types.InternalServiceError
+	assert.ErrorAs(t, err, &expectedErr)
+	assert.ErrorContains(t, err, "LeaseTaskList requires non empty task list")
+}
+
+func TestLeaseTaskList_selectErr(t *testing.T) {
+	store, db := setupNoSQLStoreMocks(t)
+
+	req := getValidLeaseTaskListRequest()
+	db.EXPECT().SelectTaskList(gomock.Any(), getExpectedTaskListFilter()).Return(nil, assert.AnError)
+	// The error is _not_ a NotFoundError
+	db.EXPECT().IsNotFoundError(assert.AnError).Return(false).Times(2)
+	db.EXPECT().IsTimeoutError(assert.AnError).Return(true).Times(1)
+
+	_, err := store.LeaseTaskList(ctx.Background(), req)
+
+	assert.Error(t, err)
+	// The function does not wrap the error, it just adds it to the message
+	assert.ErrorContains(t, err, assert.AnError.Error())
+}
+
+func TestLeaseTaskList_selectErrNotFound(t *testing.T) {
+	store, db := setupNoSQLStoreMocks(t)
+
+	req := getValidLeaseTaskListRequest()
+	db.EXPECT().SelectTaskList(gomock.Any(), getExpectedTaskListFilter()).Return(nil, assert.AnError)
+	// The error _is_ a NotFoundError
+	db.EXPECT().IsNotFoundError(assert.AnError).Return(true)
+
+	// We then expect the tasklist to be inserted
+	db.EXPECT().InsertTaskList(gomock.Any(), gomock.Any()).
+		DoAndReturn(func(ctx ctx.Context, taskList *nosqlplugin.TaskListRow) error {
+			checkTaskListRowExpected(t, taskList)
+			return nil
+		})
+
+	resp, err := store.LeaseTaskList(ctx.Background(), req)
+
+	assert.NoError(t, err)
+	checkTaskListInfoExpected(t, resp.TaskListInfo)
+}
+
+func TestLeaseTaskList_BadRenew(t *testing.T) {
+	store, db := setupNoSQLStoreMocks(t)
+
+	req := getValidLeaseTaskListRequest()
+	req.RangeID = 1 // Greater than 0, so we are trying to renew
+
+	taskListRow := getExpectedTaskListRow()
+	taskListRow.RangeID = 5 // The range ID in the DB is different from the one in the request
+
+	db.EXPECT().SelectTaskList(gomock.Any(), getExpectedTaskListFilter()).Return(taskListRow, nil)
+
+	_, err := store.LeaseTaskList(ctx.Background(), req)
+
+	var expectedErr *persistence.ConditionFailedError
+	assert.ErrorAs(t, err, &expectedErr)
+	expectedMessage := "leaseTaskList:renew failed: taskList:test-tasklist, taskListType:0, haveRangeID:1, gotRangeID:5"
+	assert.ErrorContains(t, err, expectedMessage)
```

Comment:
I think checking for error type should be better.
```
assert.ErrorAs(t, err, new(*persistence.ConditionFailedError))
```

Reply:
It was already checking the error type, but changed the check to your suggestion